File: remappedSrc/grauly/hudcompass/waypoints/Waypoint.java
package grauly.hudcompass.waypoints;


import net.minecraft.util.math.Vec3d;

public class Waypoint {
    private Vec3d waypoint;
    private String name;

    public Waypoint(Vec3d waypoint, String name) {
        this.waypoint = waypoint;
        this.name = name;
    }

    public Waypoint(float x, float y, float z, String name) {
        this.waypoint = new Vec3d(x,y,z);
        this.name = name;
    }

    public Vec3d getWaypoint() {
        return waypoint;
    }

    public String getName() {
        return name;
    }
}

File: src/client/kotlin/grauly/hudcompass/waypoint/Waypoint.kt
package grauly.hudcompass.waypoint

import grauly.hudcompass.waypoint.location.WaypointLocation
import grauly.hudcompass.rendering.waypoint.WaypointRenderState
import net.minecraft.util.Identifier

interface Waypoint {
    fun getLocation(): WaypointLocation
    fun shouldShow(dimensionId: Identifier): Boolean
    fun extractRenderState(): WaypointRenderState
}